astropy/utils/data.py

```diff
@@ -963,18 +964,39 @@ def check_free_space_in_dir(path, size):
                       f"to download a {human_file_size(size)} file")
 
 
+class _ftptlswrapper(urllib.request.ftpwrapper):
+    def init(self):
+        self.busy = 0
+        self.ftp = ftplib.FTP_TLS()
+        self.ftp.connect(self.host, self.port, self.timeout)
+        self.ftp.login(self.user, self.passwd)
+        self.ftp.prot_p()
+        _target = '/'.join(self.dirs)
+        self.ftp.cwd(_target)
+
+
+class _FTPTLSHandler(urllib.request.FTPHandler):
+    def connect_ftp(self, user, passwd, host, port, dirs, timeout):
+        return _ftptlswrapper(user, passwd, host, port, dirs, timeout,
+                              persistent=False)
+
 def _download_file_from_source(source_url, show_progress=True, timeout=None,
                                remote_url=None, cache=False, pkgname='astropy',
-                               http_headers=None):
+                               http_headers=None, ftp_tls=False):
     from astropy.utils.console import ProgressBarOrSpinner
 
     if remote_url is None:
         remote_url = source_url
     if http_headers is None:
         http_headers = {}
 
+    if ftp_tls:
+        urlopener = urllib.request.build_opener(_FTPTLSHandler())
```

Comment:
Should there be a check to raise exception if this option is given but URL is not FTP?

Reply:
I don't *think* so.  This is pretty deep into some hacky layers of `urllib`, but based on https://docs.python.org/3/library/urllib.request.html#urllib.request.build_opener it looks to me like this just *replaces* the default FTP handler with the new one.  So if the user passes in a non-ftp url a different handler takes care of it and this one is just skipped.